src/test/java/org/apache/datasketches/theta/CompactSketchTest.java

```diff
@@ -333,6 +333,55 @@ public void checkHeapCompactSketchCompact() {
     assertEquals(csk.getCurrentPreambleLongs(), 2);
   }
 
+  @Test
+  public void checkDirectCompactSketchCompact() {
+    WritableMemory wmem1, wmem2;
+    CompactSketch csk1, csk2;
+    int bytes;
+    int lgK = 6;
+
+    //empty
+    UpdateSketch sk = Sketches.updateSketchBuilder().setLogNominalEntries(lgK).build();
+    bytes = sk.getCompactBytes();
+    wmem1 = WritableMemory.allocate(bytes);
+    wmem2 = WritableMemory.allocate(bytes);
+    csk1 = sk.compact(true, wmem1); //place into memory
+    assertTrue(csk1 instanceof DirectCompactSketch);
+    csk2 = csk1.compact(true, wmem2);
+    assertTrue(csk2 instanceof DirectCompactSketch);
+
+    //single
+    sk.update(1);
+    bytes = sk.getCompactBytes();
+    wmem1 = WritableMemory.allocate(bytes);
+    wmem2 = WritableMemory.allocate(bytes);
+    csk1 = sk.compact(true, wmem1); //place into memory
+    assertTrue(csk1 instanceof DirectCompactSketch);
+    csk2 = csk1.compact(true, wmem2);
+    assertTrue(csk2 instanceof DirectCompactSketch);
+
+    //exact
+    sk.update(2);
+    bytes = sk.getCompactBytes();
+    wmem1 = WritableMemory.allocate(bytes);
+    wmem2 = WritableMemory.allocate(bytes);
+    csk1 = sk.compact(true, wmem1); //place into memory
+    assertTrue(csk1 instanceof DirectCompactSketch);
+    csk2 = csk1.compact(true, wmem2);
+    assertTrue(csk2 instanceof DirectCompactSketch);
```

Comment:
What is the significance of the additional value, when compared to a sketch containing a single value?  Considering that lgK = 6, it is clearly still in exact mode, is the intention here to see that the additional value serialises properly into memory?

Reply:
Yes. But I need to add more asserts to make the difference between these cases more clear.